libraries/lib-cloud-audiocom/ServiceConfig.cpp

```diff
@@ -11,53 +11,135 @@
 #include "ServiceConfig.h"
 #include "Languages.h"
 
-#include <rapidjson/document.h>
+#include <cassert>
 #include <stdexcept>
+#include <string_view>
+
+#include <rapidjson/document.h>
+
+#include "Prefs.h"
+#include "CodeConversions.h"
+
+
 
 namespace cloud::audiocom
 {
-std::string_view ServiceConfig::GetAPIEndpoint() const
+namespace
 {
-   return "https://api.audio.com";
+StringSetting audioComApiEndpoint {
+   L"/CloudServices/AudioCom/ApiEndpoint",
+   L"https://api.audio.com"
+};
+
+StringSetting audioComOAuthClientID {
+   L"/CloudServices/AudioCom/OAuthClientID",
+   L"1741964426607541"
+};
+
+StringSetting audioComOAuthClientSecret {
+   L"/CloudServices/AudioCom/OAuthClientSecret",
+   L"shKqnY2sLTfRK7hztwzNEVxnmhJfOy1i"
+};
+
+StringSetting audioComOAuthRedirectURL {
+   L"/CloudServices/AudioCom/OAuthRedirectURL",
+   L"https://audio.com/auth/sign-in/success"
+};
+
+StringSetting audioComOAuthLoginPage {
+   L"/CloudServices/AudioCom/OAuthLoginPage",
+   L"https://audio.com/audacity/link?clientId={auth_client_id}"
+};
+
+StringSetting audioComFinishUploadPage {
+   L"/CloudServices/AudioCom/OAuthLoginPage",
+   L"https://audio.com/audacity/upload?audioId={audio_id}&token={auth_token}&clientId={auth_client_id}"
+};
+
+StringSetting audioComAudioURL {
+   L"/CloudServices/AudioCom/AudioURL",
+   L"https://audio.com/{user_slug}/audio/{audio_slug}/edit"
+};
+
+StringSetting audioComAudioPreferredMimeType {
+   L"/CloudServices/AudioCom/PreferredMimeType",
+   L"https://audio.com/{user_slug}/audio/{audio_slug}/edit"
+};
+
+std::string Substitute(std::string pattern, std::initializer_list<std::pair<std::string_view, std::string_view>> substitutions)
+{
+   for(auto& [key, value] : substitutions)
+   {
+      auto pos = pattern.find(key);
+
+      assert(pos != std::string::npos);
+
+      if(pos != std::string::npos)
+         pattern.replace(pos, key.size(), value);
+   }
+
+   return std::move(pattern);
```

Comment:
redundant `move`

Reply:
I'm unsure that NRVO will work here, provided the argument is returned.